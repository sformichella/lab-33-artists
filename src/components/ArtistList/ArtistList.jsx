import React from 'react';
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom';
import ArtistListItem from '../ArtistListItem/ArtistListItem';
import styles from './ArtistList.css';
import { useApp } from '../../state/appContext';

export default function ArtistList({ artists, type }) {
  const { setArtistName, setRecording } = useApp();

  const onClick = name => {
    if(type === 'artist') {
      setArtistName(name);
    }

    if(type === 'recording') {
      setRecording(name);
    }
  };

  const artistsList = artists.map(artist => {
    const { id, name, image } = artist;

    return (
      <li key={id} className={styles.li}>
        <Link to={`/${type}/${id}`} onClick={() => onClick(name)}>
          <ArtistListItem name={name} image={image}/>
        </Link>
      </li>
    );
  });

  return (
    <ul data-testid="list">
      { artistsList }
    </ul>
  );
}

ArtistList.propTypes = {
  artists: PropTypes.array.isRequired,
  type: PropTypes.string.isRequired
};
