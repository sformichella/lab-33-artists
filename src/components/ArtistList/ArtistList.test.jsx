import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArtistList from './ArtistList';

describe('ArtistList component', () => {
  afterEach(() => cleanup());
  it('renders ArtistList', () => {
    const { asFragment } = render(<ArtistList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
