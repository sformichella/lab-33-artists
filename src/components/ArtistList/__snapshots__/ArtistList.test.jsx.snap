// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistList component renders ArtistList 1`] = `
<DocumentFragment>
  <ul
    data-testid="list"
  >
    <li
      class="li"
    >
      <a
        href="/artists/coolId"
      >
        Cool Artist
        <img
          alt="Cool Artist"
          class="image"
          src="cool-image.com"
        />
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
